### DISCUSSION TITLE
Representational Method Signature

### DISCUSSION THREAD

**Discussion by BigSquareHasNoEdge** [dotnet/csharplang#6519] [General]
How about this idea?
Wouldn't it be interesting if C# compiler would have this feature?

Declaration type 1:
`void On(string propertyName)PropertyChanged { }`

call:
`On("Name")PropertyChanged;`
or
`On%PropertyChanged("Name");`
or
`OnPropertyChanged("Name");`

Declaration type 2:
`void Add(int n)To(Store store){ }`

call:
`Add(1)To(myStore);`
or
`Add%To(1, myStore);`
or
`AddTo(1, myStore);`

Declaration type 3
`void Subscribe(object subscriber)To(string messageKey)Of<TPublisher> { }`

call:
`Subscribe(this)To("ProductAdded")Of<ProductViewModel>;`
or
`Subscribe%To(this, "ProductAdded")Of<ProductViewModel>;`
or
`Subscribe%To%Of<ProductViewModel>(this, "ProductAdded");`
or
`SubscribeToOf<ProductViewModel>(this, "ProductAdded");`

**Comment by siegfriedpammer** [comment]
This style is one of the things I dislike about Objective C...

**Comment by huoyaoyuan** [comment]
This can be even harder to understand for non-English speakers.
